tests/unit_tests/config/inline_loader.cpp

```diff
@@ -292,3 +292,108 @@ TEST(config, inline_loader_accepts_valid_configuration)
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_THROW(loader.load_all(), xlio_exception);
+}
+
+// Test rejecting newline character in value
+TEST(config, inline_loader_rejects_newline_in_value)
+{
+    env_setter setter("XLIO_INLINE_CONFIG", "network.protocols.ip.mtu=90\n00");
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_THROW(loader.load_all(), xlio_exception);
+}
+
+// Test rejecting carriage return in value
+TEST(config, inline_loader_rejects_carriage_return_in_value)
+{
+    env_setter setter("XLIO_INLINE_CONFIG", "network.protocols.ip.mtu=90\r00");
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_THROW(loader.load_all(), xlio_exception);
+}
+
+// Test rejecting key with trailing dot
+TEST(config, inline_loader_rejects_trailing_dot_in_key)
+{
+    env_setter setter("XLIO_INLINE_CONFIG", "core.log.=5");
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_THROW(loader.load_all(), xlio_exception);
+}
+
+// Test rejecting key with consecutive dots
+TEST(config, inline_loader_rejects_consecutive_dots_in_key)
+{
+    env_setter setter("XLIO_INLINE_CONFIG", "core..log.level=5");
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_THROW(loader.load_all(), xlio_exception);
+}
+
+// Test rejecting key with multiple consecutive dots
+TEST(config, inline_loader_rejects_multiple_consecutive_dots_in_key)
+{
+    env_setter setter("XLIO_INLINE_CONFIG", "core...log=5");
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_THROW(loader.load_all(), xlio_exception);
+}
+
+// Test accepting uppercase boolean "TRUE"
+TEST(config, inline_loader_accepts_uppercase_true)
+{
+    env_setter setter("XLIO_INLINE_CONFIG", "core.quick_init=TRUE");
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_NO_THROW(loader.load_all());
+
+    std::map<std::string, std::experimental::any> data = loader.load_all();
+    ASSERT_EQ(true, std::experimental::any_cast<bool>(data["core.quick_init"]));
+}
+
+// Test accepting uppercase boolean "FALSE"
+TEST(config, inline_loader_accepts_uppercase_false)
+{
+    env_setter setter("XLIO_INLINE_CONFIG", "core.quick_init=FALSE");
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_NO_THROW(loader.load_all());
+
+    std::map<std::string, std::experimental::any> data = loader.load_all();
+    ASSERT_EQ(false, std::experimental::any_cast<bool>(data["core.quick_init"]));
+}
+
+// Test accepting mixed case boolean "True"
+TEST(config, inline_loader_accepts_mixed_case_true)
+{
+    env_setter setter("XLIO_INLINE_CONFIG", "core.quick_init=True");
+    inline_loader loader("XLIO_INLINE_CONFIG");
+    ASSERT_NO_THROW(loader.load_all());
+
+    std::map<std::string, std::experimental::any> data = loader.load_all();
+    ASSERT_EQ(true, std::experimental::any_cast<bool>(data["core.quick_init"]));
+}
+
+// Test accepting mixed case boolean "False"
+TEST(config, inline_loader_accepts_mixed_case_false)
```

Comment:
confused now about test titles, so we accept or reject?
anyway - for bools we want to accept only lowercase